martin-mbtiles/src/tile_copier.rs

```diff
@@ -290,6 +290,45 @@ async fn open_and_detect_type(mbtiles: &Mbtiles) -> MbtResult<MbtType> {
     mbtiles.detect_type(&mut conn).await
 }
 
+pub async fn apply_mbtiles_diff(
+    src_file: PathBuf,
+    diff_file: PathBuf,
+) -> MbtResult<SqliteConnection> {
+    let src_mbtiles = Mbtiles::new(src_file)?;
+    let diff_mbtiles = Mbtiles::new(diff_file)?;
+
+    let opt = SqliteConnectOptions::new().filename(src_mbtiles.filepath());
+    let mut conn = SqliteConnection::connect_with(&opt).await?;
+    let src_type = src_mbtiles.detect_type(&mut conn).await?;
+
+    if src_type != MbtType::TileTables {
+        return Err(MbtError::InvalidDataFormat(
```

Comment:
we need a special error for this. The format is not invalid, it is simply not compatible with the given operation